gprofiler/profilers/java.py

```diff
@@ -245,8 +245,8 @@ class JavaProfiler(ProcessProfilerBase):
     JDK_EXCLUSIONS = ["OpenJ9", "Zing"]
     SKIP_VERSION_CHECK_BINARIES = ["jsvc"]
 
-    def __init__(self, frequency: int, duration: int, stop_event: Event, storage_dir: str):
-        super().__init__(frequency, duration, stop_event, storage_dir)
+    def __init__(self, frequency: int, duration: int, stop_event: Event, storage_dir: str, **profiler_kwargs):
```

Comment:
What are these for?

Reply:
So I can send ALL of the possible args to ALL of the profilers, and each profiler defines the kwargs it is actually interested in. A pretty clean way to ensure dependency injection for a factory.

Reply:
Uhh. I don't like this implicitness. It is *very* error prone (e.g non-matching strings preventing the real argument from being passed).

The `**` unpacking part is okay (since Python checks you pretty good - denies passing the same arg twice, denies passing an arg that's not accepted - as long as you don't allow extra unchecked `**`)

It reminded me this post: https://rhettinger.wordpress.com/2011/05/26/super-considered-super/ which I remember reading once (and I encourage you to!). It shows how `**kwargs` can be used in inheritance chains, while the last entry in the chain calls `super().__init__(**kwargs)` and in this case `super().__init__ == object.__init__` which expects 0 arguments - so it makes sure all arguments were really unpacked and accepted by someone along the chain. It's legit to use this way, but not without verifying.

We can discuss other ways to achieve the implicit factory you built here; otherwise let's cancel it